src/protocols/bdx/BdxTransferSession.h

```diff
@@ -165,6 +165,24 @@ class DLL_EXPORT TransferSession
      */
     void PollOutput(OutputEvent & event, System::Clock::Timestamp curTime);
 
+    /**
+     * @brief
+     *   Gets the pending output event from the transfer session in the event param passed in by the caller.
+     *   The output event may contain some data for the caller to act upon.
+     *
+     *   It is possible that consecutive calls to this method may emit different outputs depending on the state of the
+     *   TransferSession object and so we need to call this until we get an event of type - OutputEventType::kNone
+     *
+     *   Note that if the event type is kMsgToSend, the caller is expected to send the message immediately over the
+     *   exchange otherwise the caller is expected to pass the event to HandleTransferSessionOutput to handle the BDX
```

Comment:
That said, what is HandleTransferSessionOutput?  It's not related to BdxTransferSession, right?  This documentation is not making sense  to me.

Reply:
I guess i should remove the part about HandleTransferSessionOutput. its a virtual method that should be implemented by a subsclass of the TransferFacilitator to have the business logic to handle BDX messages or transfer session events.